### PR TITLE
fix: guard against infinite loops

### LINKED ISSUE #10272: RangeError
### Expected Behavior

RangeError: Invalid array length at Array.push (<anonymous>) at https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:53998 at Object.Es [as useMemo] (https://insomnia-app.local/assets/Button-C8sA8rsq.js:6:21421) at r.useMemo (https://insomnia-app.local/assets/index-BT9U50Rx.js:1:7206) at An (https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:53669) at Ht (https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:56957) at CollectionRoot (https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:56834) at wo (https://insomnia-app.local/assets/Button-C8sA8rsq.js:6:17038) at _o (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:3159) at pc (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:45008) at sc (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:39918) at Kd (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:39846) at Ul (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:39700) at Wo (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:36044) at oc (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:36855) at Vt (https://insomnia-app.local/assets/Button-C8sA8rsq.js:6:3271) at Gd (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:41438) at mn (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:40411) at ic (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:35844) at Te (https://insomnia-app.local/assets/Button-C8sA8rsq.js:1:10044) at MessagePort.lt (https://insomnia-app.local/assets/Button-C8sA8rsq.js:1:10417)

### Actual Behavior

RangeError: Invalid array length at Array.push (<anonymous>) at https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:53998 at Object.Es [as useMemo] (https://insomnia-app.local/assets/Button-C8sA8rsq.js:6:21421) at r.useMemo (https://insomnia-app.local/assets/index-BT9U50Rx.js:1:7206) at An (https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:53669) at Ht (https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:56957) at CollectionRoot (https://insomnia-app.local/assets/Modal-DA4NQuf7.js:6:56834) at wo (https://insomnia-app.local/assets/Button-C8sA8rsq.js:6:17038) at _o (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:3159) at pc (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:45008) at sc (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:39918) at Kd (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:39846) at Ul (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:39700) at Wo (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:36044) at oc (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:36855) at Vt (https://insomnia-app.local/assets/Button-C8sA8rsq.js:6:3271) at Gd (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:41438) at mn (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:40411) at ic (https://insomnia-app.local/assets/Button-C8sA8rsq.js:8:35844) at Te (https://insomnia-app.local/assets/Button-C8sA8rsq.js:1:10044) at MessagePort.lt (https://insomnia-app.local/assets/Button-C8sA8rsq.js:1:10417)

### Reproduction Steps

_No response_

### Is there an existing issue for this?

- [x] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Which sync method do you use?

- [ ] Git sync.
- [ ] Insomnia Cloud sync.
- [ ] Local only

### Additional Information

_No response_

### Insomnia Version

13.0.2

### What operating system are you using?

Windows

### Operating System Version

mac os

### Installation method

rest

### Last Known Working Insomnia version

_No response_

### PR DESCRIPTION
**Pull request by ryan-willis**
closes #10272

### CODE REVIEW

File: packages/insomnia/src/common/project.ts [hunk 1]
```diff
@@ -209,8 +209,18 @@ export async function getAllLocalFiles({ projectId }: { projectId: string }) {
   return files;
 }
 
-export const getUnsyncedRemoteWorkspaces = (remoteFiles: InsomniaFile[], workspaces: Workspace[]) =>
-  remoteFiles.filter(remoteFile => !workspaces.find(w => w._id === remoteFile.id));
+export const getUnsyncedRemoteWorkspaces = (remoteFiles: InsomniaFile[], workspaces: Workspace[]) => {
```
Comment (fiosman) [inline packages/insomnia/src/common/project.ts:212]: Thoughts about adding unit tests to this function? I noticed it has none initially. IMO since we just added more logic we should cover with a few cases.

File: packages/insomnia/src/ui/utils/dedupe-collection-items.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export function dedupeCollectionItems<T>(items: T[], getId: (item: T) => string): T[] {
```
Comment (fiosman) [inline packages/insomnia/src/ui/utils/dedupe-collection-items.ts:1]: I may be wrong but can this be simplified to just passing a number representing the `id` to dedupe?

File: packages/insomnia/src/common/project.ts (hunk 1 above)
Comment (ryan-willis) [reply packages/insomnia/src/common/project.ts:212]: There is a unit test: https://github.com/Kong/insomnia/pull/10281/changes#diff-9cf43d03a109e00c8b83c6c8050b7ff0586d14c003a3c741f1bb08960adbefdfR19

File: packages/insomnia/src/ui/utils/dedupe-collection-items.ts (hunk 2 above)
Comment (ryan-willis) [reply packages/insomnia/src/ui/utils/dedupe-collection-items.ts:1]: Can you give an example? This function just takes an array and returns a copy with duplicates (identified via the `getId` fn) removed:
```
dedupeCollectionItems(items, item => item.id)
```
it's agnostic to the shape of the items in the array:
```
dedupeCollectionItems(worspaces, wksp => wksp.workspaceId)
```

File: packages/insomnia/src/common/project.ts (hunk 1 above)
Comment (fiosman) [reply packages/insomnia/src/common/project.ts:212]: Oops, missed that! :)

File: packages/insomnia/src/ui/utils/dedupe-collection-items.ts (hunk 2 above)
Comment (fiosman) [reply packages/insomnia/src/ui/utils/dedupe-collection-items.ts:1]: I misunderstood how the function behaves; this is correct as is

**Review by fiosman — APPROVED** [review]
